packages/mint-components/src/components/sqm-referral-code/useReferralCode.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+  useEngagementMedium,
+  useMutation,
+  useParentValue,
+  useProgramId,
+  useQuery,
+  useUserIdentity,
+} from "@saasquatch/component-boilerplate";
+import { useState } from "@saasquatch/universal-hooks";
+import { gql } from "graphql-request";
+import { CopyTextViewProps } from "../views/copy-text-view";
+import {
+  ReferralCodeContext,
+  REFERRAL_CODES_NAMESPACE,
+  SET_CODE_COPIED,
+} from "../sqm-referral-codes/useReferralCodes";
+
+interface ReferralCodeProps {
+  programId?: string;
+  tooltiptext: string;
+  tooltiplifespan: number;
+  codeOverride?: string;
+}
+
+const MessageLinkQuery = gql`
+  query ($programId: ID) {
```

Comment:
I've had issues with batching and unnamed queries in the past. Prob best to give this a unique-ish name